roles/kubernetes/control-plane/vars/main.yaml

```diff
@@ -1,3 +1,3 @@
 ---
 # list of admission plugins that needs to be configured
-kube_apiserver_admission_plugins_needs_configuration: [EventRateLimit]
+kube_apiserver_admission_plugins_needs_configuration: [EventRateLimit, PodSecurity]
```

Comment:
This variable has been created to ensure that admission plugins that need to be configured, could be provided with this.
In this case, I think it is not necessary to add `PodSecurity` here because the plugin doesn't need configuration to work.
Here are some examples on how to use it:
https://github.com/kubernetes-sigs/kubespray/search?q=kube_apiserver_admission_plugins_needs_configuration

Reply:
@alegrey91  Getting this [PodSecurityConfiguration](https://github.com/kubernetes-sigs/kubespray/pull/9017/files#diff-b0903679f8b21f320860f39e46a4c69163d8eaa91eb8a9f8ec8d280fc3872de1) file created on the node is the sole reason I got into creating this PR. It's an admission plugin so I thought this was a good place to get it done. If there is a better place or if you think it's more clear to do it in other way, I'm ready to fix that. But the plugin needs the config to work.
I think this also answers your [other comment](https://github.com/kubernetes-sigs/kubespray/pull/9017#issuecomment-1182550241) about the PodSecurityConfiguration not being applied to the cluster.

The patch definitely does what's advertised - enable default PodSecurity for whole cluster and when configured, prevent insecure Pods from being spawned. Trust me, it forced me to fix quite a few apps ;).

Reply:
@cristicalin i've added a condition around the PodSecurity in here. I did it the most naive way. Not sure if it's up to standards. Please check. 
The original idea was to get the file created always, just empty when `kube_pod_security_use_default` is false.

Reply:
@Foxlik I got your point. 

So, you can leave the code in this way:
```
kube_apiserver_admission_plugins_needs_configuration: [EventRateLimit,PodSecurity]
```
You should probably rename the file [`roles/kubernetes/control-plane/templates/podsecurity.v1beta2.yaml.j2`](https://github.com/kubernetes-sigs/kubespray/pull/9017/files#diff-b0903679f8b21f320860f39e46a4c69163d8eaa91eb8a9f8ec8d280fc3872de1) to `roles/kubernetes/control-plane/templates/podsecurity.yaml.j2`.

This because the [`admission-controls.yaml.j2`](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/kubernetes/control-plane/templates/admission-controls.yaml.j2) will grab the name from the variable `kube_apiserver_admission_plugins_needs_configuration` and use it as name of the `AdmissionConfiguration` plugin list.

E.g.
```

apiVersion: apiserver.config.k8s.io/v1
kind: AdmissionConfiguration
plugins:
- name: EventRateLimit
  path: /path/to/eventratelimit.yaml
- name: PodSecurity                        # extracted from plugin variable
  path: /path/to/podsecurity.yaml          # plugin | lower
```

Ping me if you have some doubt about the explanation.
Anyway, you did a great work!